os/board/common/partitions.c

```diff
@@ -121,17 +123,21 @@ void configure_partitions(void)
 		}
 
 #if defined(CONFIG_MTD_PARTITION_NAMES)
-		if (*names) {
-			mtd_setpartitionname(mtd_part, names);
-		}
 
-		while (*names) {
-			if (*names == ',') {
+		if (*names) {
+			while (*names) {
+				if (*names == ',') {
+					part_name[index] = '\0';
+					index = 0;
+					names++;
+					break;
+				}
+				part_name[index++] = *names;
 				names++;
-				break;
 			}
-			names++;
+			mtd_setpartitionname(mtd_part, part_name);
```

Comment:
How about copy whole data into an array and use address by two indexes?

Reply:
I think two indexes need more memory. so i like it.